docs/data/data-grid/editing/editing.md

```diff
@@ -4,266 +4,408 @@ title: Data Grid - Editing
 
 # Data Grid - Editing
 
-<p class="description">The data grid has built-in edit capabilities that you can customize to your needs.</p>
+<p class="description">The data grid has built-in support for cell and row editing.</p>
 
-## Cell editing
+> ⚠️ This page refers to the new editing API, not enabled by default.
+> To use it, enable the following flag:
+>
+> ```tsx
+> <DataGrid experimentalFeatures={{ newEditingApi: true }} />
+> ```
+>
+> This additional step is required because the default editing API has a couple of issues which can only be fixed with breaking changes, only possible in v6.
+> To avoid having to wait for the next major release window, all breaking changes needed were included inside this flag.
+>
+> If you are looking for the documentation for the default editing API, visit [this page](/components/data-grid/editing-legacy/).
+> Note that it is encouraged to migrate to the new editing API since it will be enabled by default in v6.
+> Although it says "experimental", you can consider it as stable.
+
+## Making a column editable
 
-Cell editing allows editing the value of one cell at a time.
-Set the `editable` property in the `GridColDef` object to `true` to allow editing cells of this column.
+You can make a column editable by enabling the `editable` property in its [column definition](/api/data-grid/grid-col-def/):
 
 ```tsx
 <DataGrid columns={[{ field: 'name', editable: true }]} />
 ```
 
+This will enable the cell editing for the specified column.
+Note that there is also a row editing mode accessible by setting the `editMode` prop to `"row"`.
+For more information, see the dedicated [section](#row-editing).
+
+The following demo shows an example of how to make the Name column editable.
+Play with it by double-clicking or pressing <kbd class="key">Enter</kbd> in any cell from this column.
+
 {{"demo": "BasicEditingGrid.js", "bg": "inline", "defaultCodeOpen": false}}
 
 ### Start editing
 
-If a cell is editable and has focus, any of the following interactions will start the edit mode:
+These are the ways to start editing a cell (or row if `editMode="row"`):
+
+- Double-clicking a cell.
+- Pressing <kbd class="key">Enter</kbd>, <kbd class="key">Backspace</kbd> or <kbd class="key">Delete</kbd>.
+  Note that the existing content in the cell is first cleared if <kbd class="key">Backspace</kbd> or <kbd class="key">Delete</kbd> is pressed.
+- Pressing any printable key, for instance `a`, `E`, `0`, or `$`.
+- Calling `apiRef.current.startCellEditMode` passing the ID and field of the cell.
+
+  ```tsx
+  apiRef.current.startCellEditMode({ id: 1, field: 'name' });
```

Comment:
"cell editing" is the name of the feature. It could create confusion like if the feature was disabled by default and `startCellEditing` had to be called to first enable the cell editing capability, not to make a cell editable. With row editing is worse I think. AG Grid calls `startEditingCell` which states clearly that it will start editing a specific cell.

I reused "mode" because we already had this concept of "cell mode" throughout the grid, e.g. `cellMode` prop. `startCellEditMode` and `stopCellEditMode` is essentially a split of `setCellMode`. I can change if everybody agrees.

Reply:
The name feels heavy
But I agree that we should aim for clarity.
So if we don't find anything better I fine with letting the current names.

Reply:
`startCellEditing` contains the name of the feature. For row editing, `startRowEditing` is confusing since the feature is disabled (unless `editMode="row"`). I can change to `startEditingCell`.

Reply:
One thing to consider is that in #4210 we'll have a `cellModesModel` prop to control the mode of the cells without using the imperative API. The concept of "cell mode" we'll hide here will become more present with this prop. We can try other names keeping the "mode". The idea is that a cell has two modes: "view" and "edit". These methods change which mode is active. It may makes more sense to use `putCellIntoEditMode` or `changeCellToEditMode`.